core/src/main/scala/chisel3/SeqUtils.scala

```diff
@@ -26,12 +26,13 @@ private[chisel3] object SeqUtils {
     } else if (in.tail.isEmpty) {
       in.head.asUInt
     } else {
-      val lo = prefix("lo") {
-        asUInt(in.slice(0, in.length/2))
-      }.autoSeed("lo")
-      val hi = prefix("hi") {
-        asUInt(in.slice(in.length/2, in.length))
-      }.autoSeed("hi")
+      val in_lo = in.slice(0, in.length/2)
+      val in_hi = in.slice(in.length/2, in.length)
+      def wrapper(s: String, in: Seq[T]) = {
+        prefix(s) { asUInt(in) }.autoSeed(s)
+      }
+      val lo = if (in_lo.length > 1) wrapper("lo", in_lo) else asUInt(in_lo)
+      val hi = if (in_hi.length > 1) wrapper("hi", in_hi) else asUInt(in_hi)
```

Comment:
I haven't run this but I don't think it's the right solution. What happens if you do something like:

```scala
class MyModule extends RawModule {
  val a, b, c, d = IO(Input(UInt(8.W)))
  val out = IO(Output(UInt()))
  out := Cat(a + 1.U, b + 1.U, c + 1.U, d + 1.U)
}
```
[Scastie link](https://scastie.scala-lang.org/ARbrm3xURKqbU3KhTCK9Ug)

My guess is that the intermediate wires will no longer get named.

I think the correct solution is:

```suggestion
      val lo = autoNameRecursively("lo") {
         asUInt(in.slice(0, in.length/2))
       }
       val hi = autoNameRecursively("hi") {
         asUInt(in.slice(in.length/2, in.length))
       }
```
Also will need to import `chisel3.plugin. autoNameRecursively`.

`autoNameRecursively` is smart and only names nodes created within the Expression passed to it, so things created previously and passed to it (like stuff from `in` potentially), will not get named.